src/legacy/core_plugins/kibana/public/discover/np_ready/angular/context.js

```diff
@@ -68,53 +66,73 @@ getAngularModule().config($routeProvider => {
     });
 });
 
-function ContextAppRouteController(
-  $routeParams,
-  $scope,
-  AppState,
-  config,
-  $route,
-  getAppState,
-  globalState
-) {
+function ContextAppRouteController($routeParams, $scope, config, $route) {
   const filterManager = getServices().filterManager;
-  const filterStateManager = new FilterStateManager(globalState, getAppState, filterManager);
   const indexPattern = $route.current.locals.indexPattern.ip;
+  const { start, stop, appState, globalState, getAppFilters, getGlobalFilters } = getState(
+    config.get('context:defaultSize'),
+    indexPattern.timeFieldName,
+    config.get('state:storeInSessionStorage')
+  );
+  this.state = _.cloneDeep(appState.getState());
+  this.filters = _.cloneDeep([...getGlobalFilters(), ...getAppFilters()]);
```

Comment:
A lot `cloneDeep` makes me sad =(
How long will it take to get rid of all the state mutations?...

Reply:
@sulemanof  I share your sadness! And since I want to make you happy, I've reduced the amount of deepClones. However can't live without one here, would need to change the way the adding filters are implemented:
 
![Bildschirmfoto 2020-02-13 um 16 49 04](https://user-images.githubusercontent.com/463851/74513501-011af580-4f0b-11ea-8da8-483985921b95.png)

Reply:
We are going to fix it: https://github.com/elastic/kibana/issues/57677